---
title: "Maximum Mean Discrepancy in High-Dimensional Biology"
author:
  - Frederick Boehm:
      email: frederick.boehm@gmail.com
      institute: [UMMS]
      correspondence: true
institute:
  - UMMS: University of Massachusetts Medical School
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
    bookdown::pdf_document2:
      citation_package: biblatex
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
      pandoc_args:
      - --lua-filter=../templates/scholarly-metadata.lua
      - --lua-filter=../templates/author-info-blocks.lua
      - --lua-filter=../templates/pagebreak.lua
    bookdown::word_document2:
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
      pandoc_args:
      - --lua-filter=../templates/scholarly-metadata.lua
      - --lua-filter=../templates/author-info-blocks.lua
      - --lua-filter=../templates/pagebreak.lua
    bookdown::html_document2:
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
      pandoc_args:
      - --lua-filter=../templates/scholarly-metadata.lua
      - --lua-filter=../templates/author-info-blocks.lua
      - --lua-filter=../templates/pagebreak.lua
bibliography: references.bib
csl: "../templates/journal-of-archaeological-science.csl" # Insert path for the bib-style
abstract: |
  Text of abstract
keywords: |
  keyword 1; keyword 2; keyword 3
highlights: |
  These are the highlights. 
linkcolor: green
urlcolor: blue
citecolor: red
---

<!-- This is the format for text comments that will be ignored during renderings. Do not put R code in these comments because it will not be ignored. -->

<!-- With the following code you can access and display values from the yml header above. -->

Keywords: `r rmarkdown::metadata$keywords`

Highlights: `r rmarkdown::metadata$highlights`

<!-- The following code chunk defines some general settings how code chunks should behave. -->

```{r setup, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  echo = FALSE,
  comment = "#>",
  fig.path = "../figures/",
  dpi = 300
)
```

<!-- The actual document text starts here: -->

# Specific Aims

The volume and dimensionality of molecular biology data has exploded over the last twenty years. Following the sequencing of the human genome, technologies such as 
DNA sequencing, RNA sequencing, and single-cell RNA sequencing have proliferated. 
The new experimental designs and high dimension of modern biological data require parallel 
development of new statistical methods and computational tools for their analysis.
Maximum mean discrepancy (MMD) is a new statistical tool with novel mathematical properties 
[@gretton2012kernel;@gretton2006kernel;@gretton2009fast;@borgwardt2006integrating]. We propose to explore its mathematical 
and statistical properties, to apply MMD methods 
in the analysis of high-dimensional biological data, and to build user-friendly R software tools that implement MMD methods.




**Aim 1: Apply MMD to high-dimensional biological data**

Two-sample problems arise in many areas of biomedical research. These include differential gene expression, comparability of data 
from different laboratories, and classification of cancers. @borgwardt2006integrating examined performance characteristics of four multivariate two-sample tests in a collection of four biological settings. They found that MMD outperformed the other three tests. 
settings of  

**Aim 2: Create and maintain an open source R package that implements MMD methods**

Current R software for MMD methods doesn't provide significance thresholds. Gretton's 
website hosts Matlab software that implements the methods in @gretton2012kernel, @gretton2009fast 
and @gretton2006kernel. We'll implement these methods in
R and C++ before assembling the code into a well documented, user-friendly R package [@r;@eddelbuettel2011rcpp]. 






# Background and Significance


MMD is a statistical method for determining whether two high-dimensional samples arise from 
distinct distributions. Throughout this document, we use the term "sample" to refer to 
measurements on a collection of subjects from a single group. In introductory statistics 
courses, one often studies Student's t-test [@student1908probable] when discerning whether 
two samples come from the same distribution. Implicit to the t-test is the assumption that 
the two samples arise from normal distributions with a single, common variance parameter.

In high-dimensional settings, the analog of Student's t-test is Hotelling's $T^2$ test [@hotelling1931generalization]. 
Like its univariate counterpart, Hotelling's test assumes a normal (multivariate joint normal) distribution of the random 
errors and a shared covariance matrix for the two groups. Departures
from these assumptions may give rise to type I and type II errors in testing. 

To combat limitations of existing high-dimensional, two-sample tests, @gretton2006kernel proposed a kernel-based 
test for the two-sample problem. Drawing on reproducing kernel Hilbert space theory, they formulate their test as a 
measure of distance between two embeddings of probability measures in a RKHS. @gretton2006kernel then write the test 
as a maximization over a subspace $\mathcal{F}$ of their RKHS, $\mathcal{H}$ (Equation \@ref(eq:define)).

\begin{equation}
MMD[\mathcal{F}, p, q] = \sup_{f \in \mathcal{F}}\left(\mathbb{E}_xf(x) - \mathbb{E}_yf(y) \right)
  (\#eq:define)
\end{equation}




```{r, microarray, echo = FALSE}
library(magrittr)
tibble::tibble(platform = c("same", "same", "different", "different"), 
               null_hypothesis = c("accepted", "rejected", "accepted", "rejected"),
               MMD = c(100, 0, 0, 100), 
               t_test = c(100, 0, 95, 5), 
               wolfowitz = c(93, 7, 0, 100),
               smirnov = c(95, 5, 29, 71)) %>%
  gt::gt() %>%
  gt::tab_header("Microarray cross-platform comparability", 
                 label = "tab:microarray")
```


```{r, cancer, echo = FALSE}
tibble::tibble(health_status = c("same", "same", "different", "different"), 
               null_hypothesis = c("accepted", "rejected", "accepted", "rejected"),
               MMD = c(100, 0, 0, 100), 
               t_test = c(100, 0, 100, 0), 
               wolfowitz = c(97, 3, 0, 100),
               smirnov = c(98, 2, 38, 62)) %>%
  gt::gt() %>%
  gt::tab_header("Cancer diagnosis",
             label = "tab:cancer")
```






# Approach






\newpage

# Colophon

This report was generated on `r Sys.time()` using the following computational environment and dependencies: 

```{r colophon, cache = FALSE}
# which R packages and versions?
if ("devtools" %in% installed.packages()) devtools::session_info()
```

The current Git commit details are:

```{r}
# what commit is this file at? 
if ("git2r" %in% installed.packages() & git2r::in_repository(path = ".")) git2r::repository(here::here())  
```
