---
title: "Maximum Mean Discrepancy in High-Dimensional Biology"
author:
  - Frederick Boehm:
      email: frederick.boehm@gmail.com
      institute: [UMMS]
      correspondence: true
institute:
  - UMMS: University of Massachusetts Medical School
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
    bookdown::pdf_document2:
      citation_package: biblatex
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
      pandoc_args:
      - --lua-filter=../templates/scholarly-metadata.lua
      - --lua-filter=../templates/author-info-blocks.lua
      - --lua-filter=../templates/pagebreak.lua
    bookdown::word_document2:
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
      pandoc_args:
      - --lua-filter=../templates/scholarly-metadata.lua
      - --lua-filter=../templates/author-info-blocks.lua
      - --lua-filter=../templates/pagebreak.lua
    bookdown::html_document2:
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
      pandoc_args:
      - --lua-filter=../templates/scholarly-metadata.lua
      - --lua-filter=../templates/author-info-blocks.lua
      - --lua-filter=../templates/pagebreak.lua
bibliography: references.bib
csl: "../templates/journal-of-archaeological-science.csl" # Insert path for the bib-style
abstract: |
  Text of abstract
keywords: |
  keyword 1; keyword 2; keyword 3
highlights: |
  These are the highlights. 
linkcolor: green
urlcolor: blue
citecolor: red
---

<!-- This is the format for text comments that will be ignored during renderings. Do not put R code in these comments because it will not be ignored. -->

<!-- With the following code you can access and display values from the yml header above. -->

Keywords: `r rmarkdown::metadata$keywords`

Highlights: `r rmarkdown::metadata$highlights`

<!-- The following code chunk defines some general settings how code chunks should behave. -->

```{r setup, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  echo = FALSE,
  comment = "#>",
  fig.path = "../figures/",
  dpi = 300
)
```

<!-- The actual document text starts here: -->

# Specific Aims

The volume and dimensionality of molecular biology data has exploded over the last twenty years. Following the sequencing of the human genome, technologies such as 
DNA sequencing, RNA sequencing, and single-cell RNA sequencing have proliferated. 
The new experimental designs and high dimension of modern biological data require parallel 
development of new statistical methods and computational tools for their analysis.
Maximum mean discrepancy (MMD) is a new statistical tool with novel mathematical properties 
[@gretton2012kernel;@gretton2006kernel;@gretton2009fast;@borgwardt2006integrating]. We propose to explore its mathematical 
and statistical properties, to apply MMD methods 
in the analysis of high-dimensional biological data, and to build user-friendly R software tools that implement MMD methods.




**Aim 1: Apply MMD to high-dimensional biological data**

Two-sample problems arise in many areas of biomedical research. These include differential gene expression, comparability of data 
from different laboratories, and classification of cancers. @borgwardt2006integrating examined performance characteristics of four multivariate two-sample tests in a collection of four biological settings. They found that MMD outperformed the other three tests. 
settings of  

**Aim 2: Create and maintain an open source R package that implements MMD methods**

Current R software for MMD methods doesn't provide significance thresholds. Gretton's 
website hosts Matlab software that implements the methods in @gretton2012kernel, @gretton2009fast 
and @gretton2006kernel. We'll implement these methods in
R and C++ before assembling the code into a well documented, user-friendly R package [@r;@eddelbuettel2011rcpp]. 






# Background and Significance


MMD is a statistical method for determining whether two high-dimensional samples arise from 
distinct distributions. Throughout this document, we use the term "sample" to refer to 
measurements on a collection of subjects from a single group. In introductory statistics 
courses, one often studies Student's t-test [@student1908probable] when discerning whether 
two samples come from the same distribution. Implicit to the t-test is the assumption that 
the two samples arise from normal distributions with a single, common variance parameter.

In high-dimensional settings, the analog of Student's t-test is Hotelling's $T^2$ test [@hotelling1931generalization]. 
Like its univariate counterpart, Hotelling's test assumes a normal (multivariate joint normal) distribution of the random 
errors and a shared covariance matrix for the two groups. Departures
from these assumptions may give rise to type I and type II errors in testing. 

## MMD and its Definition

To combat limitations of previous high-dimensional, two-sample tests, @gretton2006kernel proposed a kernel-based 
test for the two-sample problem. Drawing on reproducing kernel Hilbert space (RKHS) theory, they formulate their test as a 
measure of distance between embeddings of probability measures in a RKHS. @gretton2006kernel then write the test 
as a maximization over a subspace $\mathcal{F}$ of their RKHS, $\mathcal{H}$ (Equation \@ref(eq:define)).

\begin{equation}
MMD[\mathcal{F}, p, q] = \sup_{f \in \mathcal{F}}\left(\mathbb{E}_xf(x) - \mathbb{E}_yf(y) \right)
  (\#eq:define)
\end{equation}

By embedding probability measures in a Hilbert space, and a RKHS in particular, @gretton2012kernel have the option of using a 
rich set of analytic tools and theory. They lean heavily on the "kernel trick", which characterizes the "reproducing" nature of the 
RKHS $\mathcal{H}$. The Riesz representation theorem for Hilbert spaces states that there is a feature map, $\phi(x) = k(x, \cdot)$ 
such that $f(x) = \langle f, \phi(x)\rangle_{\mathcal{H}}$ for any $f \in \mathcal{H}$. 
@gretton2012kernel then define an embedding for probability distribution $p$ as an element $\mu_p \in \mathcal{H}$ satisfying $\mathbb{E}_xf = \langle f, \mu_p\rangle_{\mathcal{H}}$.

## MMD Properties

## @borgwardt2006integrating Results

In aiming to understand and characterize the performance of MMD-based statistical hypothesis tests, we begin by considering the 
striking results from @borgwardt2006integrating. @borgwardt2006integrating obtained two data sets from published breast cancer studies. 
Both data sets featured gene expression measurements on 2,166 common genes, but the measurements were obtained from two different 
microarray platforms. @borgwardt2006integrating emphasized that, if there were no differences between the measurements on the two 
platforms, then the two data sets could be analyzed together. A joint analysis, when appropriate, would likely lead to greater 
statistical power to infer meaningful expression signatures for breast cancer. After scaling and centering expression measurements, 
they examined four tests. For the MMD-based test, they chose a Gaussian kernel with standard deviation set to 20. 
@borgwardt2006integrating compared the MMD-based test to standard multivariate tests including Hotelling's $T^2$ test, the 
Friedman-Rafsky multivariate Kolmogorov-Smirnov test, and the Friedman-Rafsky Wald-Wolfowitz test. 

In the first of two analyses, the study design consisted of randomly choosing 25 subjects' vectors of expression measurements from each 
sample (*i.e.*, each platform). That is, the gene expression vectors of length 2,166 were used. The two resulting subsamples consisted 
of 25 by 2166 matrices. To these pairs of subsamples, @borgwardt2006integrating applied the four tests. They repeated the entire 
procedure 100 times. They recorded the number of instances in which each of the four multivariate tests rejected the null hypothesis 
that the two subsamples arose from the same platform. Table \@ref(tab:microarray) tabulates the counts for the 100 instances of each 
test. 

The second of two analyses had a similar study design, except that instead of choosing 25 subjects from each of two platforms, 
@borgwardt2006integrating first chose one of the two platforms. They then chose two groups of 25 from the chosen platform and compared 
the resulting subsamples. Again, they repeated the procedure 100 times and recorded the results in Table \@ref(tab:microarray).




```{r, microarray, echo = FALSE}
library(magrittr)
tibble::tibble(platform = c("same", "same", "different", "different"), 
               null_hypothesis = c("accepted", "rejected", "accepted", "rejected"),
               MMD = c(100, 0, 0, 100), 
               t_test = c(100, 0, 95, 5), 
               wolfowitz = c(93, 7, 0, 100),
               smirnov = c(95, 5, 29, 71)) %>%
  gt::gt() %>%
  gt::tab_header("Microarray cross-platform comparability", 
                 label = "tab:microarray")
```

The results in Table \@ref(tab:microarray) display perfect accuracy for the MMD-based test: it correctly identified 100 of 100 
subsamples chosen from different platforms and 100 of 100 subsamples chosen from the same platform. The Friedman-Rafsky tests performed 
at an intermediate level, while the Hotelling's $T^2$ test performed poorly. Specifically, the Hotelling's $T^2$ test incorrectly 
called 95 of 100 cases when the two subsamples were chosen from different platforms.  




```{r, cancer, echo = FALSE}
tibble::tibble(health_status = c("same", "same", "different", "different"), 
               null_hypothesis = c("accepted", "rejected", "accepted", "rejected"),
               MMD = c(100, 0, 0, 100), 
               t_test = c(100, 0, 100, 0), 
               wolfowitz = c(97, 3, 0, 100),
               smirnov = c(98, 2, 38, 62)) %>%
  gt::gt() %>%
  gt::tab_header("Cancer diagnosis",
             label = "tab:cancer")
```


The remarkable performance of the MMD-based tests in these two high-dimensional settings deserves further study. It is from this perspective that we propose research to more completely characterize and understand the MMD-based test.







# Approach

## Approach for Aim 1

With the goal of enhancing understanding of the MMD-based test in high dimensions, we will undertake the following research plan. 
First, we'll identify existing R code that implements Hotelling's $T^2$ test and the two Friedman-Rafsky tests. 
By achieving **Aim 2**, we'll have R implementations of the MMD-based tests.   

## Approach for Aim 2






\newpage

# Colophon

This report was generated on `r Sys.time()` using the following computational environment and dependencies: 

```{r colophon, cache = FALSE}
# which R packages and versions?
if ("devtools" %in% installed.packages()) devtools::session_info()
```

The current Git commit details are:

```{r}
# what commit is this file at? 
if ("git2r" %in% installed.packages() & git2r::in_repository(path = ".")) git2r::repository(here::here())  
```
